### DISCUSSION TITLE
TIFFs with more than 550MB having large number of pages

### DISCUSSION THREAD

**Discussion by nishars** [ImageMagick/ImageMagick#2191] [Help]
Hi,

I have to loop through all the pages in a multi-page TIFF and there can be max 800 pages to find all the colored pages.
TIFF can have both B/W and colored pages.

I am using ReaderSetting with FrameIndex = <page number> to load single page at a time.

For smaller images it works fine, but I tried with a 580MB image it took almost 2 hours to complete.
Memory consumption was around 8GB.
The TIFF is LZW compressed, 1685 x 2185 pixels with 200DPI resolution and 24 bit depth.

I am using Magick.NET-Q8-AnyCPU nuget package in .Net core application hosted on windows and also testing it on Linux docker container.

Please suggest any resolutions to make it faster.

**Comment by dlemstra** [comment]
Can you demonstrate this is a sample project?

**Reply by nishars** [reply]
I am using the below method to loop through all the pages in tiff file, identify if it is colored and reduce its resolution, size and depth:
I observed that memory consumption is increasing since all the 800 images are added in ImageCollection and not disposed until Write().
Is there any other way to do this?

 
        public void ResaveFileUsingMagickPageSettings(string tiffFile)
        {
            using (var imageCollection = new MagickImageCollection())
            {
                var readSettings = new MagickReadSettings
                {
                    FrameCount = 1,
                    FrameIndex = 0
                };

                using (var tif = new MagickImageCollection(tiffFile, readSettings))
                {
                    tif.Ping(tiffFile);
                    for (int i = 0; i < tif.Count; i++)
                    {
                        MagickReadSettings settings = new MagickReadSettings()
                        {
                            FrameIndex = i,
                            FrameCount = 1,
                        };
                        settings.SetDefine(MagickFormat.Tif, "ignore-layers", true);

                        try
                        {
                            using (MagickImage image = new MagickImage(tiffFile, settings))
                            {
                                if (image.ColorSpace == ImageMagick.ColorSpace.sRGB)
                                {
                                    image.Settings.Compression = CompressionMethod.LZW;

                                    Density density = new Density(100, 100);
                                    image.Density = density;
                                    image.Resize(850, 1100);
                                    image.ColorType = ColorType.Palette;
                                    image.Depth = 8;
                                }

                                imageCollection.Add(image.Clone());
                            }
                        }
                        catch (MagickException ex)
                        {

                        }
                    }
                    imageCollection.Write(@"C:\Temp\USAA_3P_1_1.tif");
                    //imageCollection.Write(@"/app/TiffImages/Temp/USAA_3P_1_1.tif");

                }
            }
        }
